### ISSUE TITLE
Update approach to pricing

### ISSUE THREAD

**Issue by alexfreska** [SiaFoundation/renterd#1303]
Set allowance, then derive storage, upload, and download cost from the allowance. To derive the the values:
- multiply the allowance by 2
- set each of upload, download, and storage each to a ratio/percent of the result
- 20% upload, 40% download, 40% storage

Reasoning:
The point is that people use the Pricing in the UI as the amount of money they want to pay. But that's not quite right. e.g. I can pay $2 per TB upload if I upload to a $1 and a $3 host. Setting the max price to $2 would be counter productive since it creates unnecessary churn. In reality we probably want a user who expects $2 per TB to set the max to $4 per TB.

**Comment by alexfreska** [comment]
@ChrisSchinnerl does the actual allowance value / remaining allowance value actually factor into host selection or does it just act as a hard limit?

**Comment by alexfreska** [comment]
@ChrisSchinnerl similar to the current estimates, this calculation also probably needs to adjust for estimated storage, download, upload values?

**Comment by ChrisSchinnerl** [comment]
> @ChrisSchinnerl just wondering: does the actual allowance value / remaining allowance value actually factor into host selection or does it just act as a hard limit?

Yes, the allowance is taken into account when computing the score of a host that we use for the host selection. e.g. if you want to spend $10 per month on 10 hosts we use $1 per host as the baseline for the pricing score we give a host. So we penalise a host if the expected amount of traffic/storage can't be paid for using $1.

> @ChrisSchinnerl similar to the current estimates, this calculation also probably needs to adjust for estimated storage, download, upload values?

Changing the expected values is part of the advanced settings right? So in that case I'd leave it up to the user to make sure the values are sane.
